apps/api/views/chat.py

```diff
@@ -231,32 +293,47 @@ def chat_start_session(request):
         )
 
     team = experiment.team
-    if request.user.is_authenticated:
-        user = request.user
-        participant_id = user.email
-        if remote_id != participant_id:
-            # Enforce this for authenticated users
-            # Currently this only happens if the chat widget is being hosted on the same OCS instance as the bot
-            return Response({"error": "Remote ID must match your email address"}, status=status.HTTP_400_BAD_REQUEST)
-        remote_id = ""
-    else:
+
+    if is_embedded:
         user = None
-        participant_id = None
+        participant_id = remote_id if remote_id else embed_participant_id
+        platform = ChannelPlatform.EMBEDDED_WIDGET
+        api_channel = experiment_channel
+        # Skip public API access checks for embedded widgets
 
-    access_response = check_experiment_access(experiment, participant_id)
-    if access_response:
-        return access_response
+    else:
+        platform = ChannelPlatform.API
+        api_channel = ExperimentChannel.objects.get_team_api_channel(team)
+
+        if request.user.is_authenticated:
+            user = request.user
+            participant_id = user.email
+            if remote_id != participant_id:
+                return Response(
+                    {"error": "Remote ID must match your email address"}, status=status.HTTP_400_BAD_REQUEST
+                )
+            remote_id = ""
+        else:
+            user = None
+            participant_id = None
+
+        access_response = check_experiment_access(experiment, participant_id)
+        if access_response:
+            return access_response
 
     # Create or get participant
     if user is not None:
         participant, created = Participant.objects.get_or_create(
             identifier=participant_id,
             team=team,
-            platform=ChannelPlatform.API,
+            platform=platform,
             defaults={"user": user, "remote_id": remote_id},
         )
     else:
-        participant = Participant.create_anonymous(team, ChannelPlatform.API, remote_id)
+        participant = Participant.create_anonymous(team, platform, remote_id)
+        if is_embedded:
+            participant.identifier = participant_id
+            participant.save(update_fields=["identifier"])
```

Comment:
Rather than doing this you could pass the prefix to the method and let it handle the rest.

```python
participant = Participant.create_anonymous(team, platform, remote_id, prefix="embed")
```